packages/controls/src/breakpoints/breakpoints.ts

```diff
@@ -161,3 +175,48 @@ export function mergeResponsiveValues<A>(
     },
   })) as DeviceOverride<A>[]
 }
+
+export function findNextFallback<V>(
+  breakpoints: Breakpoints,
+  value: ResponsiveValue<V>,
+  deviceId: Breakpoint['id'],
+  activeDeviceId: Breakpoint['id'],
+  fallbackStrategy?: FallbackStrategy<V>,
+): Breakpoint | null {
+  const deviceOverride = findBreakpointOverride(
+    breakpoints,
+    value.filter((v) => v.deviceId !== activeDeviceId),
+    deviceId,
+    fallbackStrategy,
+  )
+
+  return deviceOverride != null
+    ? getBreakpoint(breakpoints, deviceOverride.deviceId)
+    : null
+}
+
+export const getBreakpointMediaQuery = (breakpoint: Breakpoint): string => {
+  const parts = ['@media only screen']
+
+  if (breakpoint.minWidth != null) {
+    parts.push(`(min-width: ${breakpoint.minWidth}px)`)
+  }
+
+  if (breakpoint.maxWidth != null) {
+    parts.push(`(max-width: ${breakpoint.maxWidth}px)`)
+  }
+
+  return parts.join(' and ')
+}
+
+export const getViewportStyle = (
+  breakpoints: Breakpoints,
+  deviceId: Breakpoint['id'],
+): Viewport<string | number> => {
```

Comment:
Moved from https://github.com/makeswift/makeswift/pull/947/files#diff-98cfbae694c1ec54b413804d485fd52c751a1e706e56312369ea0b9e7f336e2dL17, fixed the return type to reflect that `getBreakpoint` throws rather than returning `null` / `undefined` when the device ID is not found in the breakpoints.